src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java

```diff
@@ -177,6 +177,11 @@ StateBackend getStateBackend() {
     return stateBackend;
   }
 
+  @Override
+  public void reloadStateFromBackend() throws IOException {
+    getStateBackend().loadOrCreateGlobalState();
```

Comment:
This call will only retrieve the new `GlobalStateInfo` from the backend. This needs to be used to build a new `ImmutableState`, recycling the `IndexStateManager`s from the current state. It might also be good to reload the index state for the previously existing indices as well.

The `updateStartedIndices` method can be used to start indices that should be started.

Once this is done, the current immutable state can be replaced by the new one.